### PR TITLE
Build .NETCoreApp source gens in allconfigurations build

### LINKED ISSUE #100346: CI doesn't build `net9.0` targets in source generators.
A while back we added `net9.0` targets to the source generator projects so that they support nullability annotations. These aren't being used by the product but help with local development. We discovered with https://github.com/dotnet/runtime/pull/100345 that these targets aren't being built by CI, so it's easy for them to be broken.

cc @ViktorHofer

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/runtime/issues/100346

### CODE REVIEW

File: src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs [hunk 1]
```diff
@@ -5621,7 +5621,7 @@ private static string DescribeLoop(RegexNode node, RegexMethod rm)
 
         private static string ToHexStringNoDashes(byte[] bytes) =>
 #if NET
-            Convert.ToHexString(bytes);
+            Convert.ToHexString(bytes);kk
```
Comment (ViktorHofer) [inline src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs:5624]: Temp change for CI

Comment (ericstj) [reply src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs:5624]: Assuming you just had this here to prove it was building in CI.   Please revert before checking in.

File: src/libraries/oob.proj [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
   <ItemGroup Condition="'$(BuildTargetFramework)' != '$(NetCoreAppCurrent)' or '$(BuildTargetFramework)' == ''">
     <ProjectReference Include="oob-all.proj" SkipGetTargetFrameworkProperties="true" />
+    <ProjectReference Include="oob-gen.proj" />
```
Comment (ericstj) [inline src/libraries/oob.proj:21]: nit: this references all library generators, not just OOB ones.  You could name it `all-gen.proj`.

**Review by ericstj — APPROVED** [review]

File: src/libraries/oob.proj (hunk 2 above)
Comment (ViktorHofer) [reply src/libraries/oob.proj:21]: It's the same with "oob-all.proj" which also references everything. I felt that it's better to follow the existing naming schema.

**Comment by ViktorHofer** [comment]
Failure is correctly reported on both Windows and Unix in the allconfigurations leg. Reverting the code change now.

> src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs(5629,9): error CS1585: (NETCORE_ENGINEERING_TELEMETRY=Build) Member modifier 'private' must precede the member type and name

**Pull request merged**